File: _1.16.5/src/main/java/cn/enaium/noexpensive/callback/impl/AnvilTakeOutputCallbackImpl.java
package cn.enaium.noexpensive.callback.impl;

import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.nbt.NbtList;

/**
 * @author Enaium
 */
public class AnvilTakeOutputCallbackImpl implements AnvilTakeOutputCallback {
    @Override
    public void interact(ItemStack output) {
        if (!output.equals(ItemStack.EMPTY)) {
            NbtCompound nbtCompound = output.getOrCreateSubTag("display");
            final NbtList list = nbtCompound.getList("Lore", 8);
            for (int i = 0; i < list.size(); i++) {
                if (list.getString(i).contains("container.repair.cost")) {
                    list.remove(i);
                    break;
                }
            }
        }
    }
}

File: modern/_1.16.5/src/main/kotlin/cn/enaium/noexpensive/callback/impl/AnvilTakeOutputCallbackImpl.kt
package cn.enaium.noexpensive.callback.impl

import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack

/**
 * @author Enaium
 */
abstract class AnvilTakeOutputCallbackImpl : AnvilTakeOutputCallback {
    override fun interact(output: ItemStack, player: PlayerEntity) {
        condition(player) && return
        val nbtCompound = output.getOrCreateSubTag("display")
        val list = nbtCompound.getList("Lore", 8)
        for (i in list.indices) {
            if (list.getString(i).contains("container.repair.cost")) {
                list.removeAt(i)
                break
            }
        }
    }

    abstract fun condition(player: PlayerEntity): Boolean
}